components/hybrid_engine/src/observer/snapshot.rs

```diff
@@ -36,11 +37,17 @@ impl HybridSnapshotObserver {
 }
 
 impl SnapshotObserver for HybridSnapshotObserver {
-    fn on_snapshot(&self, ctx: SnapshotContext, sequence_number: u64) -> Box<dyn ObservedSnapshot> {
+    fn on_snapshot(
```

Comment:
Based on previous discussion, shall we remove this `HybridSnapshotObserver`? Or you decided to do it in another PR?

Reply:
I want to do it in another PR, because it requires more tests.